lib/progress.ts

```diff
@@ -58,8 +59,51 @@ export enum EventType {
    * Another type of detail message
    */
   DEBUG = 'debug',
+
+  /**
+   * When a shell command is executed. Emits the the command line arguments given to
+   * the subprocess as a string upon shell execution.
+   *
+   * Only emitted when subprocessOutputDestination is set to 'publish'
+   */
+  SHELL_OPEN = 'shell_open',
+
+  /**
+   * Text output from a shell command
+   *
+   * Only emitted when subprocessOutputDestination is set to 'publish'
+   */
+  SHELL_DATA = 'shell_data',
+
+  /**
+   * When a shell command closes. Emits the the command line arguments given to
+   * the subprocess as a string upon shell closure.
+   *
+   * Only emitted when subprocessOutputDestination is set to 'publish'
+   */
+  SHELL_CLOSE = 'shell_close',
 }
 
+/**
+ * A helper function to convert shell events to asset progress events
+ * @param event a shell event
+ * @returns an {@link EventType}
+ */
+export function shellEventToEventType(event: ShellEventType): EventType {
+  switch (event) {
+    case 'open':
+      return EventType.SHELL_OPEN;
+    case 'close':
+      return EventType.SHELL_CLOSE;
+    case 'data_stdout':
+      return EventType.SHELL_DATA;
+    case 'data_stderr':
+      return EventType.SHELL_DATA;
```

Comment:
Do we lose the ability to tell whether it's stdout or stderr data? I would think that's a rather important distinction for consumers...?

(Sorry I didn't pick up on this before)

Reply:
Yeah since this part is new I honestly figured that it wouldn't be relevant to distinguish between the 2 but I can always break it out in to `shell_stderr` and `shell_stdout` if we want

Reply:
went ahead and did that